src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -21,19 +21,43 @@ public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
 				if (DispatcherProvider.SetCurrent(provider))
 					svc.CreateLogger<Dispatcher>()?.LogWarning("Replaced an existing DispatcherProvider with one from the service provider.");
 
-				return Dispatcher.GetForCurrentThread()!;
+				var dispatch = Dispatcher.GetForCurrentThread();
+
+				return dispatch ?? svc.GetRequiredService<ApplicationDispatcher>().AppDispatcher;
```

Comment:
I added the coalescing here, because we don't want to break users that are retrieving the scoped `IDispatcher` from the root service provider on a background thread. Because we are no longer saturating the `IDispatcher` on the root provider we need to account for this.